Arrowgene.Ddon.GameServer/Handler/QuestQuestProgressHandler.cs

```diff
@@ -89,6 +89,28 @@ public override void Handle(GameClient client, StructurePacket<C2SQuestQuestProg
 
                     // Remove the quest data from the party object
                     partyQuestState.CompleteQuest(questId);
+
+                    if (quest.QuestType == QuestType.Main)
+                    {
+                        var leaderCommonId = client.Party.Leader.Client.Character.CommonId;
+                        // TODO: Eventually handle all types of quests and for all members
```

Comment:
Don't the other quests also work if you save the progress the same way? Besides allowing them to be played again

Reply:
Sure, but I didn't test it out yet which is why I have the TODO :)